backend/app/store/search/bleve.go

```diff
@@ -0,0 +1,242 @@
+package search
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"time"
+
+	"github.com/blevesearch/bleve/v2"
+
+	analyzerCustom "github.com/blevesearch/bleve/v2/analysis/analyzer/custom"
+	analyzerStandard "github.com/blevesearch/bleve/v2/analysis/analyzer/standard"
+
+	// not all currently supported locales are supported by bleve, import only those that are
+	langAr "github.com/blevesearch/bleve/v2/analysis/lang/ar"
+	langDe "github.com/blevesearch/bleve/v2/analysis/lang/de"
+	langEn "github.com/blevesearch/bleve/v2/analysis/lang/en"
+	langEs "github.com/blevesearch/bleve/v2/analysis/lang/es"
+	langFi "github.com/blevesearch/bleve/v2/analysis/lang/fi"
+	langFr "github.com/blevesearch/bleve/v2/analysis/lang/fr"
+	langIt "github.com/blevesearch/bleve/v2/analysis/lang/it"
+	langRu "github.com/blevesearch/bleve/v2/analysis/lang/ru"
+
+	"github.com/blevesearch/bleve/v2/analysis/token/lowercase"
+	"github.com/blevesearch/bleve/v2/mapping"
+	"github.com/umputun/remark42/backend/app/store"
+)
+
+var analyzerNameMapping = map[string]string{
+	"standard": analyzerStandard.Name,
+	"ar":       langAr.AnalyzerName,
+	"de":       langDe.AnalyzerName,
+	"en":       langEn.AnalyzerName,
+	"es":       langEs.AnalyzerName,
+	"fi":       langFi.AnalyzerName,
+	"fr":       langFr.AnalyzerName,
+	"it":       langIt.AnalyzerName,
+	"ru":       langRu.AnalyzerName,
+}
+
+type bleveEngine struct {
+	index     bleve.Index
+	indexPath string
+}
+
+// Index adds set of comments to the index
+func (b *bleveEngine) Index(comments []store.Comment) error {
+	batch := b.index.NewBatch()
+	for _, comment := range comments {
+		key := DocumentKey{Locator: comment.Locator, ID: comment.ID}
+		keyBytes, err := json.Marshal(key)
+		if err != nil {
+			return fmt.Errorf("can't marshal key %+v: %w", key, err)
+		}
+
+		doc := newBleveDoc(comment)
+		err = batch.Index(string(keyBytes), doc)
+		if err != nil {
+			return fmt.Errorf("can't add to indexing batch: %w", err)
+		}
+	}
+	err := b.index.Batch(batch)
+	if err != nil {
+		return fmt.Errorf("index error: %w", err)
+	}
+
+	return nil
+}
+
+// Search performs search request
+func (b *bleveEngine) Search(req *Request) (*Result, error) {
+	bQuery := bleve.NewQueryStringQuery(req.Query)
+	bReq := bleve.NewSearchRequestOptions(bQuery, req.Limit, req.Offset, false)
+
+	switch {
+	case req.SortBy == "":
+		bReq.SortBy([]string{"-time"})
+	case req.SortBy == "time" || req.SortBy == "-time" || req.SortBy == "+time":
+		bReq.SortBy([]string{req.SortBy})
+	default:
```

Comment:
Please add test covering that case as well.

Reply:
Added to search_test https://github.com/umputun/remark42/pull/757/commits/3e4515b0cbecc0927cfc3a47defbee1f3dd3675e#diff-f933ab083e2ed57c319100cfb2bc25f3fc8493387c6b2fd5531600d3150870aaR109

And also there's case in rest tests https://github.com/umputun/remark42/pull/757/commits/3e4515b0cbecc0927cfc3a47defbee1f3dd3675e#diff-f09c2502b9be06e17728dceb28af8a6a4c70e1db2cadfa1fd62b9d91af7f1897R253